src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -73,10 +65,12 @@ export class DefaultTelemetryService implements TelemetryService {
         return this._telemetryLogger
     }
 
-    public async start(): Promise<void> {
+    public async start(cachePath: string = this.persistFilePath): Promise<void> {
         // TODO: `readEventsFromCache` should be async
-        this._eventQueue.push(...DefaultTelemetryService.readEventsFromCache(this.persistFilePath))
-        this._endOfCache = this._eventQueue[this._eventQueue.length - 1]
+        const cacheEvents = DefaultTelemetryService.readEventsFromCache(cachePath)
+        if (cacheEvents.length > 0) {
+            await this.flushRecords(cacheEvents)
```

Comment:
This will increase total extension activation time, but I don't know by how much. VS Code only reports synchronous activation time, not async. If we started emitting a bunch of metrics this could be pretty bad. Maybe benchmark how long this line takes with a cache filled with 100 metrics.

Reply:
That is why we didn't eager-flush here. 

For reference, https://github.com/aws/aws-toolkit-vscode/pull/926 added `this._eventQueue.push(...)` to fix a different issue (though I don't think this PR regresses it).

Reply:
We could just not await the call in that case. Eagerly flushing resolves a lot of complexity around handling the cache.

Reply:
> but without the await we should make sure the 'publisher' can handle 'concurrent' flushes

That's why I didn't suggest removing `await`. Also removing `await` doesn't gain anything really because most of our startup cost is CPU-bound.  So I suggest keeping `await`.

Reply:
The only other way I can think of doing this is if we set up a list of promises to be run following activation (which...technically, becomes a part of activation, but wouldn't block reaching the extension's functionality). Would something like that sound good? We could use it beyond this as well.